client/common.go

```diff
@@ -0,0 +1,37 @@
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Client is the programmatic interface to the marketplace server API.
+type Client struct {
```

Comment:
Maybe we should _not_ export this. This was we can make sure, all instances are created through `NewClient`.

Reply:
That would would trigger a `go lint` warning: https://github.com/golang/lint/issues/210. It would also suppress `Client` from godoc.

Reply:
Ah, I know there was a reason this isn't a common pattern. As much as I like to reduce the API surface, it's probably not worth the hassle with `go lint`.

But suppressing it from godoc is a plus in my opinion, because it's not important what the fields are.